File: wwwwswe/CHAPTER-03/실전문제/3-1.js
// n을 lst의 가장 큰 수부터 나눠서 갯수를 구함, 가장 큰 단위 ~> 가장 작은 단위까지
// 갖고 있는 동전 중에서 큰 단위가 항상 작은 단위의 배수이므로 작은 단위의 동전들을 종합해 다른 해가 나올 수 없기 때문에 가능

function countChange(n) {
    const lst = [500, 100, 50, 10];
    let count = 0;
  
    for (let coin of lst) {
      count += Math.floor(n / coin);
      n %= coin;
    }
  
    return count;
  }
  
  console.log(countChange(1260)) // 6

File: wwwwswe/CHAPTER-03/실전문제/python/3-1.py
def calculate_change(n):
    coin_types = [500, 100, 50, 10]
    count = 0
    for coin in coin_types:
        count += n // coin # 해당 동전이 사용될 수 있는 개수를 더함
        n %= coin # 해당 동전을 사용한 후 남은 잔돈을 구한다
    return count

print(calculate_change(1260)) # 6